node/node_resharding.go

```diff
@@ -1,300 +0,0 @@
-		node.nextShardState.proposeTime = time.Now().Add(15 * time.Second)
-	}
-	// TODO ek – this should be done from replaying beaconchain once
-	//  beaconchain sync is fixed
-	err = node.Beaconchain().WriteShardState(
-		receivedEpoch, epochShardState.ShardState)
-	if err != nil {
-		return ctxerror.New("cannot store shard state", "epoch", receivedEpoch).
-			WithCause(err)
-	}
-	return nil
-}
-
-/*
-func (node *Node) transitionIntoNextEpoch(shardState types.State) {
-	logger = logger.New(
-		"blsPubKey", hex.EncodeToString(node.Consensus.PubKey.Serialize()),
-		"curShard", node.Blockchain().ShardID(),
-		"curLeader", node.Consensus.IsLeader())
-	for _, c := range shardState {
-		utils.Logger().Debug().
-			Uint32("shardID", c.ShardID).
-			Str("nodeList", c.NodeList).
-         Msg("new shard information")
-	}
-	myShardID, isNextLeader := findRoleInShardState(
-		node.Consensus.PubKey, shardState)
-	logger = logger.New(
-		"nextShard", myShardID,
-		"nextLeader", isNextLeader)
-
-	if myShardID == math.MaxUint32 {
-		getLogger().Info("Somehow I got kicked out. Exiting")
-		os.Exit(8) // 8 represents it's a loop and the program restart itself
-	}
-
-	myShardState := shardState[myShardID]
-
-	// Update public keys
-	var publicKeys []*bls.PublicKey
-	for idx, nodeID := range myShardState.NodeList {
-		key := &bls.PublicKey{}
-		err := key.Deserialize(nodeID.BlsPublicKey[:])
-		if err != nil {
-			getLogger().Error("Failed to deserialize BLS public key in shard state",
-				"idx", idx,
-				"error", err)
-		}
-		publicKeys = append(publicKeys, key)
-	}
-	node.Consensus.UpdatePublicKeys(publicKeys)
-	//	node.DRand.UpdatePublicKeys(publicKeys)
-
-	if node.Blockchain().ShardID() == myShardID {
-		getLogger().Info("staying in the same shard")
-	} else {
-		getLogger().Info("moving to another shard")
-		if err := node.shardChains.Close(); err != nil {
-			getLogger().Error("cannot close shard chains", "error", err)
-		}
-		restartProcess(getRestartArguments(myShardID))
-	}
-}
-*/
-
-func findRoleInShardState(
-	key *bls.PublicKey, state shard.State,
-) (shardID uint32, isLeader bool) {
-	keyBytes := key.Serialize()
-	for idx, shard := range state {
-		for nodeIdx, nodeID := range shard.NodeList {
-			if bytes.Compare(nodeID.BlsPublicKey[:], keyBytes) == 0 {
-				return uint32(idx), nodeIdx == 0
-			}
-		}
-	}
-	return math.MaxUint32, false
-}
-
-func restartProcess(args []string) {
-	execFile, err := getBinaryPath()
-	if err != nil {
-		utils.Logger().Error().
-			Err(err).
-			Str("file", execFile).
-			Msg("Failed to get program path when restarting program")
-	}
-	utils.Logger().Info().
-		Strs("args", args).
-		Strs("env", os.Environ()).
-		Msg("Restarting program")
-	err = syscall.Exec(execFile, args, os.Environ())
-	if err != nil {
-		utils.Logger().Error().
-			Err(err).
-			Msg("Failed to restart program after resharding")
-	}
-	panic("syscall.Exec() is not supposed to return")
-}
-
-func getRestartArguments(myShardID uint32) []string {
```

Comment:
The restart part will be needed when real resharding is added.

Reply:
Removed because not used at moment & I see resharding being done in own package - but can add back